src/domTree.js

```diff
@@ -344,8 +336,164 @@ class symbolNode {
+        this.height = height || "";
+        this.x = x || "";
+        this.viewBoxWidth = viewBoxWidth || 0;
+        this.viewBoxHeight = viewBoxHeight || 0;
+        this.align = align || "";
+    }
+
+    toNode() {
+        const svgNS = "http://www.w3.org/2000/svg";
+        const node = document.createElementNS(svgNS, "svg");
+
+        if (this.x.length > 0) {
+            node.setAttribute("x", this.x);
+        }
+
+        if (this.width.length > 0) {
+            node.setAttribute("width", this.width);
+        }
+
+        if (this.width.length > 0) {
+            node.setAttribute("height", this.height + "em");
+        }
+
+        if (this.viewBoxWidth > 0) {
+            node.setAttribute("viewBox",
+                `0 0 ${this.viewBoxWidth} ${this.viewBoxHeight}`);
+        }
+
+        if (this.align.length > 0) {
+            const par = (this.align === "none" ? "none" : this.align + " slice");
+            node.setAttribute("preserveAspectRatio", par);
+        }
+
+        if (this.children.length > 0) {
+            for (let i = 0; i < this.children.length; i++) {
+                node.appendChild(this.children[i].toNode());
+            }
+        }
+        return node;
+    }
+
+    toMarkup() {
+        let markup = "<svg";
+
+        if (this.x.length > 0) {
+            markup += ` x='${this.x}'`;
+        }
+
+        if (this.width.length > 0) {
+            markup += ` width="${this.width}'`;
+        }
+
+        if (this.width.length > 0) {
+            markup += " height='" + this.height + "em'";
+        }
+
+        if (this.viewBoxWidth > 0) {
+            markup +=
+                ` viewBox='0 0 ${this.viewBoxWidth} ${this.viewBoxHeight}'`;
+        }
+
+        if (this.align.length > 0) {
+            const align = (this.align === "none" ? "none" :
+                this.align + " slice");
+            markup += ` preserveAspectRatio='${align}'`;
+        }
+
+        markup += ">";
+
+        for (let i = 0; i < this.children.length; i++) {
+            markup += this.children[i].toMarkup();
+        }
+
+        markup += "</svg>";
+
+        return markup;
+
+    }
+}
+
+class pathNode {
+    constructor(pathName, alternate) {
+        this.pathName = pathName || "";
+        this.alternate = alternate || "";  // Used only for tall \sqrt
+    }
+
+    toNode() {
+        const svgNS = "http://www.w3.org/2000/svg";
+        const node = document.createElementNS(svgNS, "path");
+
+        if (this.pathName !== "sqrtTall") {
+            node.setAttribute("d", svgGeometry.path[this.pathName]);
+        } else {
+            node.setAttribute("d", this.alternate);
+        }
+
+        return node;
+    }
+
+    toMarkup() {
+        if (this.pathName !== "sqrtTall") {
+            return `<path d='${svgGeometry.path[this.pathName]}'/>`;
+        } else {
+            return `<path d='${this.alternate}'/>`;
+        }
+    }
+}
+
+class lineNode {
+    constructor(x1, y1, x2, y2, strokeWidth) {
```

Comment:
This could be simplified to (not critical though):
```
constructor(x1 = 0, y1 = 0, x2 = 0, y2 = 0, strokeWidth = 0) {
    Object.assign({x1, y1, x2, y2, strokeWidth});
}
```

Reply:
A single `attributes` arg might be simpler in the constructor. But it also might be more complex in `stretchy.js` I'll look at it.

Reply:
In this case I probably wouldn't bother with a single `attributes` arg even though it deviates from the other constructors.  The reason is that x1, y1, x2, y2 is a well-known ordering of args and you're never going to skip one... it's probably all or none.